src/audio/module_adapter/module_adapter.c

```diff
@@ -176,7 +176,7 @@ int module_adapter_prepare(struct comp_dev *dev)
 		comp_err(dev, "module_adapter_prepare() error %x: module prepare failed",
 			 ret);
 
-		return -EIO;
+		return ret;
```

Comment:
Nice, I actually commented on this exact code earlier today. If you want you can make it even more compact as
```
if (ret != PPL_STATUS_PATH_STOP)
	comp_err(dev, "module_adapter_prepare() error %x: module prepare failed", ret);
return ret;
```
but that's up to you